### PR TITLE
[Debt] Use php enums with graphql schema

### LINKED ISSUE #2400: Use native PHP Enums with the GraphQL schema
## Description
We currently define enums in our graphql.schema file then we have to define them a second time in the ApiEnums file so that we have constants we can use in factories, seeders, etc.   This would be about defining them just once.

Ref: https://lighthouse-php.com/6/the-basics/types.html#native-php-definition

## Acceptance Criteria
- [ ] Enums used in the schema are defined only once, in PHP.

### PR DESCRIPTION
**Pull request by JamesHuf**
🤖 Resolves #2400

## 👋 Introduction

This moves the definition for GraphQL enums to PHP so we only need to define them once.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Confirm tests pass.
2. Confirm queries, mutations, factories, seeders and such all still work.

### CODE REVIEW

File: api/app/Models/PoolCandidate.php [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 use App\Http\Resources\UserResource;
 use App\Observers\PoolCandidateObserver;
+use App\Providers\CandidateExpiryFilter;
+use App\Providers\CandidateSuspendedFilter;
+use App\Providers\PoolCandidateStatus;
+use App\Providers\PublishingGroup;
 use Carbon\Carbon;
 use Database\Helpers\ApiEnums;
```
Comment (petertgiles) [inline api/app/Models/PoolCandidate.php:12]: unused import

File: api/app/Providers/GraphQLServiceProvider.php [hunk 2]
```diff
@@ -3,11 +3,641 @@
 namespace App\Providers;
 
 use App\GraphQL\Operators\PostgreSQLOperator;
+use GraphQL\Type\Definition\Description;
+use GraphQL\Type\Definition\EnumType;
 use Illuminate\Support\ServiceProvider;
+use Nuwave\Lighthouse\Schema\TypeRegistry;
 use Nuwave\Lighthouse\WhereConditions\Operator;
 
+enum Language: string
```
Comment (petertgiles) [inline api/app/Providers/GraphQLServiceProvider.php:12]: Having all the enums declared directly in the App\Providers namespace and in this service provider file doesn't seem quite clean to me. (This page)[https://www.itsolutionstuff.com/post/how-to-use-enum-in-laravelexample.html] suggests app/Enums which seems reasonable to me.  Alternatively, you could leave these definitions in the regular ApiEnums file.  Maybe worth a quick discussion in scrum today.

File: api/database/helpers/ApiEnums.php [hunk 3]
```diff
@@ -5,44 +5,6 @@
...
-            self::OPERATIONAL_REQUIREMENT_TRANSPORT_EQUIPMENT,
-            self::OPERATIONAL_REQUIREMENT_DRIVERS_LICENSE,
-            self::OPERATIONAL_REQUIREMENT_OVERTIME_SCHEDULED,
-            self::OPERATIONAL_REQUIREMENT_OVERTIME_SHORT_NOTICE,
-            self::OPERATIONAL_REQUIREMENT_OVERTIME_OCCASIONAL,
-            self::OPERATIONAL_REQUIREMENT_OVERTIME_REGULAR,
-        ];
-    }
-
     const LANGUAGE_ABILITY_ENGLISH = 'ENGLISH';
```
Comment (petertgiles) [inline api/database/helpers/ApiEnums.php:8]: Wow, there's quite a bit left in here.  Out of scope for this PR but we should probably review these and move them to a "deprecated enums" file or something.

File: api/database/helpers/ApiEnums.php
Comment (petertgiles) [inline api/database/helpers/ApiEnums.php:1]: Don't forget there's a second ApiEnums file for the directive forms that needs work, too.

File: api/graphql/schema.graphql
Comment (petertgiles) [inline api/graphql/schema.graphql:1]: Don't forget there's a second graphql file for the directive forms that needs work, too.

**Review by petertgiles — CHANGES_REQUESTED** [review]
Nice, thanks for taking this one!  :bow: It's so nice to have these consolidated.  I'd like to open up the question of where these new enums should live in scrum today.

File: api/app/Providers/GraphQLServiceProvider.php (hunk 2 above)
Comment (petertgiles) [reply api/app/Providers/GraphQLServiceProvider.php:12]: It looks like [the BenSampo library](https://github.com/BenSampo/laravel-enum#basic-usage) that covered enums before they came natively to PHP also uses App\Enums.

File: api/app/Models/PoolCandidate.php (hunk 1 above)
Comment (JamesHuf) [reply api/app/Models/PoolCandidate.php]: Removed with [849ee42](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7923/commits/849ee4285d5994748aba12e705bf5d7c7f3b2210).

File: api/app/Providers/GraphQLServiceProvider.php (hunk 2 above)
Comment (JamesHuf) [reply api/app/Providers/GraphQLServiceProvider.php]: These have been moved to a new app\Enums folder with [849ee42](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7923/commits/849ee4285d5994748aba12e705bf5d7c7f3b2210).

File: api/database/helpers/ApiEnums.php
Comment (JamesHuf) [reply api/database/helpers/ApiEnums.php:1]: Not anymore. 😄

File: api/graphql/schema.graphql
Comment (JamesHuf) [reply api/graphql/schema.graphql:1]: These have been updated with [000a3ef](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7923/commits/000a3ef2ef2ac528a1e734d01bcf65a81812e099) and [0fed08a](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7923/commits/0fed08a57b094418a925e88089e4baf96d050f52).

**Review by petertgiles — APPROVED** [review]

**Comment by petertgiles** [comment]
I love this so much.  So, so much cleaner.  :smile:

**Pull request merged**